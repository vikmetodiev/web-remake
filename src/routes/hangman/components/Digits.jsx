import React from 'react';
import styled from 'styled-components';

const Block = styled.ul`
  padding: 0;
  display: ${props => (props.newGame ? 'block' : 'none')};
  list-style-type: none;
  margin: 5px 0;
  li {
    cursor: pointer;
    margin-right: 10px;
    margin-top:5px;
    padding: 3px;
    text-transform: uppercase;
    font-size: 22px;
    display: inline-block;
    border: 1px solid black;
    border-radius:4px;
    width:27px;
    background-color:white

  }
  li:hover {
    background-color: tomato;
  }
  li[disabled] {
    text-decoration: line-through;
    opacity: 0.4;
    pointer-events: none;
  }
`;

const letters = 'abcdefghijklmnopqrstuvwxyz'.split('');
const numbers = '0123456789'.split('');

const Digits = props => (
  <div>
    <Block {...props}>
      {letters.map(letter => (
        <li
          key={letter}
          className="digit"
          disabled={props.guesses.includes(letter) ? true : false}
          onClick={props.onClick}
        >
          {letter}
        </li>
      ))}
    </Block>
    <Block {...props}>
      {numbers.map(number => (
        <li
          key={number}
          className="digit"
          disabled={props.guesses.includes(number) ? true : false}
          onClick={props.onClick}
        >
          {number}
        </li>
      ))}
    </Block>
  </div>
);

export default Digits;
