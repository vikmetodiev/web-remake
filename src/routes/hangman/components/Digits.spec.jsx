import React from 'react';

import toJson from 'enzyme-to-json';
import Digits from './Digits';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('Digits', () => {
  it('should render the component', () => {
    const wrapper = shallow(<Digits guesses="abc" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
