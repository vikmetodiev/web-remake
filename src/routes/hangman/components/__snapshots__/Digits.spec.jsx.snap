// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Digits should render the component 1`] = `
<div>
  <styled.ul
    guesses="abc"
  >
    <li
      className="digit"
      disabled={true}
      key="a"
    >
      a
    </li>
    <li
      className="digit"
      disabled={true}
      key="b"
    >
      b
    </li>
    <li
      className="digit"
      disabled={true}
      key="c"
    >
      c
    </li>
    <li
      className="digit"
      disabled={false}
      key="d"
    >
      d
    </li>
    <li
      className="digit"
      disabled={false}
      key="e"
    >
      e
    </li>
    <li
      className="digit"
      disabled={false}
      key="f"
    >
      f
    </li>
    <li
      className="digit"
      disabled={false}
      key="g"
    >
      g
    </li>
    <li
      className="digit"
      disabled={false}
      key="h"
    >
      h
    </li>
    <li
      className="digit"
      disabled={false}
      key="i"
    >
      i
    </li>
    <li
      className="digit"
      disabled={false}
      key="j"
    >
      j
    </li>
    <li
      className="digit"
      disabled={false}
      key="k"
    >
      k
    </li>
    <li
      className="digit"
      disabled={false}
      key="l"
    >
      l
    </li>
    <li
      className="digit"
      disabled={false}
      key="m"
    >
      m
    </li>
    <li
      className="digit"
      disabled={false}
      key="n"
    >
      n
    </li>
    <li
      className="digit"
      disabled={false}
      key="o"
    >
      o
    </li>
    <li
      className="digit"
      disabled={false}
      key="p"
    >
      p
    </li>
    <li
      className="digit"
      disabled={false}
      key="q"
    >
      q
    </li>
    <li
      className="digit"
      disabled={false}
      key="r"
    >
      r
    </li>
    <li
      className="digit"
      disabled={false}
      key="s"
    >
      s
    </li>
    <li
      className="digit"
      disabled={false}
      key="t"
    >
      t
    </li>
    <li
      className="digit"
      disabled={false}
      key="u"
    >
      u
    </li>
    <li
      className="digit"
      disabled={false}
      key="v"
    >
      v
    </li>
    <li
      className="digit"
      disabled={false}
      key="w"
    >
      w
    </li>
    <li
      className="digit"
      disabled={false}
      key="x"
    >
      x
    </li>
    <li
      className="digit"
      disabled={false}
      key="y"
    >
      y
    </li>
    <li
      className="digit"
      disabled={false}
      key="z"
    >
      z
    </li>
  </styled.ul>
  <styled.ul
    guesses="abc"
  >
    <li
      className="digit"
      disabled={false}
      key="0"
    >
      0
    </li>
    <li
      className="digit"
      disabled={false}
      key="1"
    >
      1
    </li>
    <li
      className="digit"
      disabled={false}
      key="2"
    >
      2
    </li>
    <li
      className="digit"
      disabled={false}
      key="3"
    >
      3
    </li>
    <li
      className="digit"
      disabled={false}
      key="4"
    >
      4
    </li>
    <li
      className="digit"
      disabled={false}
      key="5"
    >
      5
    </li>
    <li
      className="digit"
      disabled={false}
      key="6"
    >
      6
    </li>
    <li
      className="digit"
      disabled={false}
      key="7"
    >
      7
    </li>
    <li
      className="digit"
      disabled={false}
      key="8"
    >
      8
    </li>
    <li
      className="digit"
      disabled={false}
      key="9"
    >
      9
    </li>
  </styled.ul>
</div>
`;
